lib/cppapi/AsyncTimer.cc

```diff
@@ -62,9 +67,9 @@ handleTimerEvent(TSCont cont, TSEvent event, void *edata)
 }
 } // namespace
 
-AsyncTimer::AsyncTimer(Type type, int period_in_ms, int initial_period_in_ms)
+AsyncTimer::AsyncTimer(Type type, int period_in_ms, int initial_period_in_ms, TSThreadPool thread_pool)
```

Comment:
If `thread_pool` had a default of the `TSThreadPool` value of `ET_NET` then existing plugins would compile and run as they do now without functional change.

Reply:
That is, `TS_THREAD_POOL_DEFAULT`.

Reply:
I accomplish this by having multiple constructors, see the header file changes below.